ngraph_bridge/ngraph_builder.cc

```diff
@@ -2531,58 +2496,33 @@ static Status TranslateUnpackOp(const Node* op,
 
   ng::Output<ng::Node> ng_input;
   TF_RETURN_IF_ERROR(GetInputNode(ng_op_map, op, 0, ng_input));
-
-  ng::Shape input_shape = ng_input.get_shape();
-  size_t input_rank = input_shape.size();
-
   int32 tf_axis;
   TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "axis", &tf_axis));
-  auto unpack_axis = tf_axis;
-  if (unpack_axis == -1) {
-    unpack_axis = input_rank - 1;
-  }
-
-  int32 tf_num;
-  TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "num", &tf_num));
-  int num_outputs = tf_num;
-
-  std::vector<int64> output_shape;
-  for (size_t i = 0; i < input_rank; ++i) {
-    if ((int)i != unpack_axis) {
-      output_shape.push_back(input_shape[i]);
-    }
-  }
-
-  ng::AxisVector ng_axis_order;
-  for (size_t i = 0; i < input_rank; i++) {
-    ng_axis_order.push_back(i);
-  }
-
-  std::vector<size_t> lower_bound_vec(input_rank, 0);
-  std::vector<size_t> upper_bound_vec(input_rank);
-
-  for (size_t i = 0; i < input_rank; i++) {
-    upper_bound_vec[i] = input_shape[i];
-  }
+  int32 num_outputs;
+  TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "num", &num_outputs));
 
+  auto input_shape = ng_input.get_shape();
+  auto rank = input_shape.size();
   for (int i = 0; i < num_outputs; ++i) {
-    lower_bound_vec[unpack_axis] = i;
-    upper_bound_vec[unpack_axis] = i + 1;
-    auto lower_bound = ConstructNgNode<opset::Constant>(
-        op->name(), ng::element::i64, ng::Shape{lower_bound_vec.size()},
-        lower_bound_vec);
-    auto upper_bound = ConstructNgNode<opset::Constant>(
-        op->name(), ng::element::i64, ng::Shape{upper_bound_vec.size()},
-        upper_bound_vec);
+    std::vector<int64_t> begin(rank, 0);
+    std::vector<int64_t> end(rank, 0);
+    begin[tf_axis] = i;
+    end[tf_axis] = i + 1;
+    auto ng_begin = ConstructNgNode<opset::Constant>(
+        op->name(), ng::element::i64, ng::Shape{begin.size()}, begin);
+    auto ng_end = ConstructNgNode<opset::Constant>(op->name(), ng::element::i64,
+                                                   ng::Shape{end.size()}, end);
+    std::vector<int64_t> begin_mask(rank, 1);
+    begin_mask[tf_axis] = 0;
+    std::vector<int64_t> end_mask(rank, 1);
+    end_mask[tf_axis] = 0;
+    std::vector<int64_t> new_axis_mask(rank, 0);
+    std::vector<int64_t> shrink_axis_mask(rank, 0);
+    shrink_axis_mask[tf_axis] = 1;
     auto slice = ConstructNgNode<opset::StridedSlice>(
-        op->name(), ng_input, lower_bound, upper_bound, std::vector<int64_t>{},
-        std::vector<int64_t>{});
-    auto ng_shape = ConstructNgNode<opset::Constant>(
-        op->name(), ng::element::u64, ng::Shape{output_shape.size()},
-        output_shape);
-    auto reshaped =
-        ConstructNgNode<opset::Reshape>(op->name(), slice, ng_shape, false);
-    SaveNgOp(ng_op_map, op->name(), reshaped);
+        op->name(), ng_input, ng_begin, ng_end, begin_mask, end_mask,
+        new_axis_mask, shrink_axis_mask);
```

Comment:
For Unpack, is there no need to use Squeeze as was suggested?

Reply:
No.